client/src/main/scala/io/chrisdavenport/circuit/http4s/client/CircuitedClient.scala

```diff
@@ -1,13 +1,138 @@
 package io.chrisdavenport.circuit.http4s.client
 
 import io.chrisdavenport.circuit.CircuitBreaker
+import cats.syntax.all._
 import cats.effect._
+import cats.effect.concurrent._
 import org.http4s._
 import org.http4s.client._
-
+import cats._
+import scala.concurrent.duration._
 object CircuitedClient {
 
   def apply[F[_]: Bracket[*[_], Throwable]](cr: CircuitBreaker[F])(c: Client[F]): Client[F] = 
-    Client{req: Request[F] => Resource(cr.protect(c.run(req).allocated))}
+    generic(Function.const[CircuitBreaker[F], Request[F]](cr))(c)
+
+
+  def byKey[F[_]](
+    maxFailures: Int,
+    resetTimeout: FiniteDuration,
+    exponentialBackoffFactor: Double = 1,
+    maxResetTimeout: Duration = Duration.Inf,
+    modifications: CircuitBreaker[F] => CircuitBreaker[F] = {x: CircuitBreaker[F] => x},
+    shouldFail: (Request[F], Response[F]) => Boolean = defaultShouldFail[F](_, _)
```

Comment:
I don't love the Boolean here.  `true` is good or `true` is failure?  I think the latter.